tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+import os
+import re
+import subprocess
+from typing import List
+from optparse import OptionParser
+
+class NodeMaintenance:
+    '''
+    Class to keep track of project, zones, and search terms for nodes, as well as results from gcloud queries
+    '''
+    def __init__(self, project, zones, regex):
+        self.project = project
+        res = subprocess.run("gcloud projects describe {}".format(self.project), shell=True, capture_output=True, text=True)
```

Comment:
nit. you can use f-string instead of `format`:

```py
f"gcloud projects describe {self.project}"
```

Reply:
Done.